lxd/auth/drivers/openfga.go

```diff
@@ -113,6 +116,83 @@ func (e *embeddedOpenFGA) load(ctx context.Context, identityCache *identity.Cach
 	return nil
 }
 
+// GetViewableProjects accepts a list of permissions and returns a list of projects that a member of a group with these permissions is able to view.
+func (e *embeddedOpenFGA) GetViewableProjects(ctx context.Context, permissions []api.Permission) ([]string, error) {
+	// Use a different context from the given context.
+	// Why? Because the OpenFGA datastore will attempt to load a cache that is optimised for permission checking an individual request.
+	// In this case we are asking the datastore to enumerate projects based on contextual tuples. The cache is not useful for this, so we don't
+	// want to trigger loading it.
+	var cancel context.CancelFunc
+	deadline, ok := ctx.Deadline()
+	if ok {
+		ctx, cancel = context.WithTimeout(context.Background(), time.Until(deadline))
+	} else {
+		ctx, cancel = context.WithTimeout(context.Background(), 5*time.Second)
+	}
+
+	defer cancel()
+
+	// Construct a list objects request using a dummy identity as a member of a dummy group.
```

Comment:
I was just thinking in:

https://github.com/canonical/lxd/pull/16547/commits/91d26d11f5837f56e27aed98e2de2108b2b60265#r2362665840

Is this the same as `lxc project list`, but the only problem is we have an identity there, whereas in the case of group management we don't have an identity yet. Then I thought I wonder if we could split apart `projectsGet()` and feed it a psuedo identity via the request context to save implementing part of `projectsGet` here.

Reply:
Unfortunately that wouldn't work. The access checking in getting a permission checker only uses groups that have been defined in the database. Whereas this function doesn't require the group to be defined yet, allowing it to be used for validation.